packages/turso-sync/src/database.rs

```diff
@@ -0,0 +1,95 @@
+use std::path::PathBuf;
+
+use hyper_rustls::{HttpsConnector, HttpsConnectorBuilder};
+use hyper_util::{client::legacy::connect::HttpConnector, rt::TokioExecutor};
+
+use crate::{
+    database_inner::{DatabaseInner, Rows},
+    errors::Error,
+    filesystem::tokio::TokioFilesystem,
+    sync_server::turso::{TursoSyncServer, TursoSyncServerOpts},
+    Result,
+};
+
+/// [Database] expose public interface for synced database from [DatabaseInner] private implementation
+///
+/// This layer also serves a purpose of "gluing" together all component for real use,
+/// because [DatabaseInner] abstracts things away in order to simplify testing
+pub struct Database(DatabaseInner<TursoSyncServer, TokioFilesystem>);
```

Comment:
I was thinking that we would have something like:

```rust
struct Session {
    pub database: turso::Database,
    pub sync: Sync,
}
```

That way we don't need to re-implement the Turso connection and statement APIs in the sync package.